### PR TITLE
Bump mlflow to 2.15.0

### PR DESCRIPTION
**Pull request by christianversloot**
## Describe changes
This PR bumps `mlflow` to its most recent version, `2.15.0`. I ran our test pipeline using `zenml==0.63.0` against `mlflow==2.15.0` and it succeeded:

image
image

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

**Review by avishniakov — APPROVED** [review]
Thanks for contributing!

**Comment by christianversloot** [comment]
@avishniakov No worries! Worthwhile to bump this PR to `2.15.1` before merging? https://github.com/mlflow/mlflow/releases/tag/v2.15.1

**Comment by htahir1** [comment]
@christianversloot go for it!

**Comment by avishniakov** [comment]
@christianversloot , we had a chat internally, what about making `"mlflow>=2.1.1,<3",` and keeping it like that until something breaks?

**Comment by christianversloot** [comment]
@avishniakov fine with me! Do your tests always work with the latest version? :)

File: src/zenml/integrations/mlflow/__init__.py
Comment (htahir1) [inline src/zenml/integrations/mlflow/__init__.py:36]:
```suggestion
        "mlflow>=2.1.1,<3",
```

**Review by htahir1 — CHANGES_REQUESTED** [review]
@christianversloot How bout trying this? We'll merge this if it works now

**Comment by avishniakov** [comment]
> @avishniakov fine with me! Do your tests always work with the latest version? :)

Yes, with the latest installable (a bit of a disclaimer if the latest-latest will not be installable due to the dependencies clash)

**Comment by christianversloot** [comment]
@htahir1 @avishniakov Sure, good idea and nice that it works, I have a rather busy schedule today so will likely be able to adapt second part of the afternoon, if you need it before feel free to push a commit yourself :)

**Review by htahir1 — APPROVED** [review]

**Pull request merged**
